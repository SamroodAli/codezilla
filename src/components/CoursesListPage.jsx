import { FlexGrid, FlexGridItem } from 'baseui/flex-grid';
import PropTypes from 'prop-types';
import VideoItem from './VideoItem';

const CoursesListPage = ({ setCurrentVideo, courses }) => (
  <FlexGrid
    flexGridColumnCount={[1, 2, 2, 4]}
  >
    {courses.map((video) => (
      <FlexGridItem
        key={video.etag}
        backgroundColor="mono300"
        display="flex"
        alignItems="center"
        justifyContent="center"
      >
        <VideoItem
          video={video.snippet}
          link={video.etag}
          key={video.etag}
          onClick={() => { setCurrentVideo(video); }}
        />
      </FlexGridItem>
    ))}
  </FlexGrid>
);

CoursesListPage.propTypes = {
  courses: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  setCurrentVideo: PropTypes.func.isRequired,
};

export default CoursesListPage;
