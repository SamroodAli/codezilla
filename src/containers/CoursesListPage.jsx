import PropTypes from 'prop-types';
import useActions from '../hooks/useActions';
import CoursesListPage from '../components/CoursesListPage';

const CoursesListPageContainer = ({ courses }) => {
  const { setCurrentVideo } = useActions();
  return <CoursesListPage courses={courses} setCurrentVideo={setCurrentVideo} />;
};

CoursesListPageContainer.propTypes = {
  courses: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default CoursesListPageContainer;
