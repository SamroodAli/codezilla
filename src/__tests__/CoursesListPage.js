import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import CoursesListPage from '../components/CoursesListPage';

describe('Testing CoursesListPage', () => {
  it('CoursesListPage is defined', async () => {
    const courses = render(<CoursesListPage courses={[]} />);
    expect(courses).toBeDefined();
  });

  it('CoursesListPage snapshot testing', () => {
    const tree = renderer.create(<CoursesListPage courses={[]} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
