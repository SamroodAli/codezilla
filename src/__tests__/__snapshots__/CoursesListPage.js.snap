// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CoursesListPage with redux CoursesListPage snapshot testing 1`] = `
<div
  className=""
  data-baseweb="flex-grid"
/>
`;
